hr_timesheet_hours_billed/models/hours_billed.py

```diff
@@ -0,0 +1,45 @@
+from odoo import api, fields, models
+
+
+class AccountAnalyticLine(models.Model):
+    _inherit = "account.analytic.line"
+
+    approved = fields.Selection(
+        selection=[("no", "No"), ("yes", "Yes")],
+        default="no",
+        required=True,
+    )
+    approved_uid = fields.Many2one(
+        "res.users",
+        string="Approved by",
+        index=True,
+        tracking=True,
+        compute="_compute_approved_uid",
+        store=True,
+    )
+    approved_date = fields.Date(
+        copy=False, compute="_compute_approved_date", store=True
+    )
+    unit_amount_billed = fields.Float(string="Hours Billed")
+
+    # set approved_uid if approved state change to 'yes'
+    @api.depends("approved")
+    def _compute_approved_uid(self):
+        for record in self:
+            if record.approved == "yes":
+                record.approved_uid = self.env.user
+
+    # set approved_date if approved state change to 'yes'
+    @api.depends("approved")
+    def _compute_approved_date(self):
+        for record in self:
+            if record.approved == "yes":
+                record.approved_date = fields.Datetime.today()
+
+    # override create method to initialize unit_amount_billed
+    @api.model_create_multi
+    def create(self, vals_list):
+        for vals in vals_list:
+            if not vals.get("unit_amount_billed"):
```

Comment:
can we use the _prepare_analytic_line method?

Reply:
I'm sorry, but I don't understand why to use it in this task.
Can you give me more details or similar examples?

Reply:
It seems to me that we can't use it. In our situation, we need to check that the field unit_amount_billed is not empty, but this field is in another model and we can't use this field in the account_move model